### PR TITLE
feat: migrate Polls plugin to new plugin infra

### PR DESCRIPTION
**Pull request by UncleBill**
part of #3066

### CODE REVIEW

File: packages/maskbook/src/plugins/Polls/base.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { Plugin } from '@masknet/plugin-infra'
+import { PLUGIN_ID, PLUGIN_ICON, PLUGIN_NAME, PLUGIN_DESCRIPTION } from './constants'
+
+export const base: Plugin.Shared.Definition = {
+    ID: PLUGIN_ID,
+    icon: PLUGIN_ICON,
+    name: { fallback: PLUGIN_NAME },
+    description: { fallback: PLUGIN_DESCRIPTION },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Polls/base.ts:9]: And this plugin is contributed by the community too.

File: packages/maskbook/src/plugins/Polls/SNSAdaptor/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    DecryptedInspector: function Comp(props) {
+        const metadata = PollMetadataReader(props.message.meta)
+        if (!metadata.ok) return null
+        return <PollsInPost {...props} />
+    },
+    CompositionDialogMetadataBadgeRender: new Map([
+        [POLL_META_KEY_1, (meta: PollMetaData) => `a poll of '${meta.question}' from ${meta.sender}`],
+    ]),
+    CompositionDialogEntry: PollCompositionEntry,
+    GlobalInjection() {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Polls/SNSAdaptor/index.tsx:26]: Why you'd keep GlobalInjection?

File: packages/maskbook/src/plugins/Polls/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { registerPlugin } from '@masknet/plugin-infra'
+import { base } from './base'
+
+registerPlugin({
+    ...base,
+    SNSAdaptor: {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Polls/index.ts:6]: IIRC this plugin definitely requires a Worker entry to work.

File: packages/maskbook/src/plugins/Polls/base.ts (hunk 1 above)
Comment (UncleBill) [reply packages/maskbook/src/plugins/Polls/base.ts:9]: Any suggestion for marking the community?

Comment (Jack-Works) [reply packages/maskbook/src/plugins/Polls/base.ts:9]: I have no idea. cc @Tedko

File: packages/maskbook/src/plugins/Polls/index.ts (hunk 3 above)
Comment (UncleBill) [reply packages/maskbook/src/plugins/Polls/index.ts:6]: Is this "empty" worker correct? https://github.com/DimensionDev/Maskbook/pull/3500/commits/21b352cf9530132b9f63c8d48666d2cdf87cf240

Comment (Jack-Works) [reply packages/maskbook/src/plugins/Polls/index.ts:6]: Hmm, I found this plugin is a bit complicated than others, let me handle this one.

File: packages/maskbook/src/plugins/Polls/base.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import type { Plugin } from '@masknet/plugin-infra'
+import { PLUGIN_ID, PLUGIN_ICON, PLUGIN_NAME, PLUGIN_DESCRIPTION } from './constants'
+
+export const base: Plugin.Shared.Definition = {
+    ID: PLUGIN_ID,
+    icon: PLUGIN_ICON,
+    name: { fallback: PLUGIN_NAME },
+    description: { fallback: PLUGIN_DESCRIPTION },
+    publisher: { name: { fallback: 'xichi' }, link: 'https://github.com/xichi' },
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Polls/base.ts:9]: cc @yisiliu or @Tedko

#1578

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/Polls/base.ts (hunk 4 above)
Comment (septs) [reply packages/maskbook/src/plugins/Polls/base.ts:9]:
```suggestion
    publisher: { name: { fallback: 'XiChi' }, link: 'https://github.com/xichi' },
```

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/Polls/base.ts (hunk 4 above)
Comment (UncleBill) [reply packages/maskbook/src/plugins/Polls/base.ts]: Patched in https://github.com/DimensionDev/Maskbook/pull/3500/commits/284d1c75db76106766b0e41b6f67cf2a3cae7115

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**